### PR TITLE
gc_worker: Implement non-stream physical scan lock

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What problem does this PR solve? 

This PR changes `physical_scan_lock` to non-streaming implementation. This is used to solve the issue that TiDB cannot block TiKV's sender when TiDB's resolving speed of locks is slower than TiKV's sending speed, which may cause TiDB consumes too much memory to buffer the locks.

The old streaming physical_scan_lock is directly changed to the streaming one. Since the feature is not enabled in any revision, so it's safe not to care about the compatibility.

### What is changed and how it works?

The API `physical_scan_lock` is changed to non-stream implementation, and mergeLockScanner is updated to adapt the new implementation.

### Check List 

Tests 

 - Unit test
 - Manual test (Test result can be found in corresponding TiKV's PR: https://github.com/tikv/tikv/pull/6631)
Code changes

 - Has exported function/method change
  - Changed MockStreamClient back to private which was made public when mergeLockScanner is implementing.

Side effects

- No

Related changes

- [x] Corresponding PR in TiKV: https://github.com/tikv/tikv/pull/6631
- [x] kvproto: https://github.com/pingcap/kvproto/pull/553

### CODE REVIEW

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -1871,39 +1870,41 @@ func (s *mergeLockScanner) physicalScanLocksForStore(ctx context.Context, safePo
...
+
 		if len(resp.Error) > 0 {
 			return errors.Errorf("physical scan lock received error from store: %v", resp.Error)
 		}
 
+		if len(resp.Locks) == 0 {
+			break
+		}
+
+		nextKey = resp.Locks[len(resp.Locks)-1].Key
```
Comment (sticnarf) [inline store/tikv/gcworker/gc_worker.go]: Maybe better to append `\0` to `nextKey`?

**Comment by MyonKeminta** [comment]
@youjiali1995 @sticnarf PTAL again. thanks!

**Pull request merged**